### PR TITLE
Use Content Graph in Generate-Docs

### PR DESCRIPTION
**Pull request by RotemAmit**
## Related Issues
fixes:https://jira-hq.paloaltonetworks.local/browse/CIAC-3909

## Description
Replaced the usage of the id_set with the graph in demisto-sdk generate-docs.
The replacement is shown by:
* Adding a new flag --graph/--no-graph
* Updating the file generate_script_doc.py to use the graph.

### CODE REVIEW

File: demisto_sdk/commands/generate_docs/generate_script_doc.py [hunk 1]
```diff
@@ -67,16 +75,26 @@ def generate_script_doc(
...
+            with ContentGraphInterface() as graph:
+                update_content_graph(
+                    graph,
+                    use_git=True,
+                    output_path=graph.output_path,
+                )
+                result = graph.search(object_id=script_id)
+                if not isinstance(result, List):
+                    raise ValueError(f"The requested script {input_path} wasn't found in the graph")
+                script_object = result[0]
```
Comment (dorschw) [inline demisto_sdk/commands/generate_docs/generate_script_doc.py:90]: what if result is empty?

File: demisto_sdk/commands/init/tests/contribution_converter_test.py [hunk 2]
```diff
@@ -14,12 +14,17 @@
...
 from demisto_sdk.commands.init.contribution_converter import (
     ContributionConverter,
     get_previous_nonempty_line,
 )
 from TestSuite.contribution import Contribution
 from TestSuite.repo import Repo
+from demisto_sdk.commands.content_graph.interface import (
+    ContentGraphInterface,
+)
+from demisto_sdk.commands.content_graph.commands.create import create_content_graph
```
Comment (dantavori) [inline demisto_sdk/commands/init/tests/contribution_converter_test.py:27]: where do we need this?

**Review by dantavori — COMMENTED** [review]
looks good, let's schedule a short demo.
my only concern to consider is - where do we use the generate-docs command? will it always allow building the graph?

File: demisto_sdk/commands/init/tests/contribution_converter_test.py (hunk 2 above)
Comment (RotemAmit) [reply demisto_sdk/commands/init/tests/contribution_converter_test.py:27]: We don't I deleted it

**Review by dantavori — APPROVED** [review]

**Pull request merged**